pkgs/games/adom/default.nix

```diff
@@ -1,59 +1,81 @@
-{ stdenv, fetchurl, patchelf, zlib, libmad, libpng12, libcaca, libGLU_combined, alsaLib, libpulseaudio
-, xorg }:
+{ stdenv, fetchurl, autoPatchelfHook, system, lib
+# Used by the ASCII build.
+, ncurses5
+# Used by the graphical build only.
+, SDL2, SDL2_image, SDL2_net, SDL2_mixer, SDL2_ttf
+, luajit
+}:
 
 let
+  version = "3.3.3";
+  sources = {
+    ascii = {
+      x86_64-linux = fetchurl {
+        url = "https://www.adom.de/home/download/current/adom_linux_ubuntu_64_${version}.tar.gz";
+        sha256 = "493ef76594c1f6a7f38dcf32a76cd107fb71dd12bf917c18fdeaebcac1a353b1";
+      };
+      i686-linux = fetchurl {
+        url = "https://www.adom.de/home/download/current/adom_linux_ubuntu_32_${version}.tar.gz";
+        sha256 = "0fayy4vz4zk45i77j5nb1jx7rkl65jyfdp2wy0y9sv9p980yz0q3";
+      };
+    };
+    graphical = {
+      x86_64-linux = fetchurl {
+        url = "https://www.indiedb.com/downloads/mirror/173928/115/4eb8407f7ab0e4143212c289acfbe735/";
+        sha256 = "047q472rrr2xwkq76431i4acbfic1q9d01x45y49bbp58fsgvkgd";
+      };
+      i686-linux = fetchurl {
+        url = "https://www.indiedb.com/downloads/mirror/173927/100/af42faf2994ef88e844fb630894f80c8";
+        sha256 = "08dn1504qjwznwmzh4304igbv8fx9azw8rq70724pv9if72yja77";
+      };
+    };
+  };
 
-  inherit (xorg) libXext libX11;
+  mkAdom = type: stdenv.mkDerivation {
+    name = "adom-${type}-${version}";
 
-  lpath = "${stdenv.cc.cc.lib}/lib64:" + stdenv.lib.makeLibraryPath [
-      zlib libmad libpng12 libcaca libXext libX11 libGLU_combined alsaLib libpulseaudio];
+    src = sources.${type}.${system};
 
-in
-stdenv.mkDerivation rec {
-  name = "adom-${version}-noteye";
-  version = "1.2.0_pre23";
+    nativeBuildInputs = [ autoPatchelfHook ];
 
-  src = fetchurl {
-    url = "http://ancardia.uk.to/download/adom_noteye_linux_ubuntu_64_${version}.tar.gz";
-    sha256 = "0sbn0csaqb9cqi0z5fdwvnymkf84g64csg0s9mm6fzh0sm2mi0hz";
-  };
+    buildInputs = [ ncurses5 ] ++ lib.optional (type == "graphical") [ 
+      SDL2 SDL2_image SDL2_net SDL2_mixer SDL2_ttf
+      luajit
+    ];
+
+    unpackPhase = ''
```

Comment:
You could probably set `sourceRoot = "./adom";` instead of overriding the whole unpack phase.

Reply:
The default unpackPhase fails to unpack the files from indiedb.com, as they lack extensions. However, on trying this again I find that the URLs no longer work -- they're presumably temporary.

Not sure what to do about that. Ask the ADOM author, probably.

Reply:
Yeah those file share sites are hostile to scripted downloads. Are you sure the author permits redistribution? :) I don't see it listed in Debian or Ubuntu package sets, even though the author provides debs...

Reply:
Debian has a thing against non-free software. It might be in the non-free repo, haven't checked.

The 'downloads' page does say to contact him if there are any packaging issues for new distros, so I'll do that.